src/components/FarmGroupDetailBody/FarmGroupDetailBody.js

```diff
@@ -12,7 +12,7 @@ export default function FarmGroupDetailBody ({
   routeParams
 }) {
   const { group } = useEnsureCurrentGroup()
-  const { posts } = useEnsurePosts({ public: true, sortBy: 'updated', context: 'groups' })
+  const { posts } = (currentUser && useEnsurePosts({ public: true, sortBy: 'updated', context: 'groups' })) || { posts: [] }
```

Comment:
I would think that React would be unhappy to only sometimes run this hook, but I may be wrong. To be safe I'd probably assign this response out to something like `postsFromStore` or whatever, then ternary `postsFromStore?.posts` to return either that result or `[]`...

Reply:
Alternatively, you could change `useEnsurePosts` to return a default of an empty array for the `posts` key...